Panic in p2p

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x70 pc=0x50a1]
goroutine 142 [running]:
panic(0x4a9c80, 0xc82000e070)
    /usr/local/Cellar/go/1.6.2/libexec/src/runtime/panic.go:481 +0x3e6
main.(*Procedures).Show(0xc82000f960, 0xc8202d6000, 0xc82014b200, 0x0, 0x0)
    /Users/p1gmale0n/projects/ssf/p2p/instance.go:288 +0xe1
reflect.Value.call(0x46e240, 0x506408, 0x13, 0x53f4e8, 0x4, 0xc8204ffee8, 0x3, 0x3, 0x0, 0x0, ...)
    /usr/local/Cellar/go/1.6.2/libexec/src/reflect/value.go:435 +0x120d
reflect.Value.Call(0x46e240, 0x506408, 0x13, 0xc8204ffee8, 0x3, 0x3, 0x0, 0x0, 0x0)
    /usr/local/Cellar/go/1.6.2/libexec/src/reflect/value.go:303 +0xb1
net/rpc.(*service).call(0xc820051d80, 0xc820051800, 0xc8204ec180, 0xc8200a0500, 0xc8201864c0, 0x3c3240, 0xc8202d6000, 0x16, 0x3c31e0, 0xc82014b200, ...)
    /usr/local/Cellar/go/1.6.2/libexec/src/net/rpc/server.go:383 +0x1c2
created by net/rpc.(*Server).ServeCodec
    /usr/local/Cellar/go/1.6.2/libexec/src/net/rpc/server.go:477 +0x49d
```

--- a/instance.go
+++ b/instance.go
@@ -285,26 +285,34 @@ func (p *Procedures) Show(args *RunArgs, resp *Response) error {
 		resp.ExitCode = 0
 		if exists {
 			if args.IP != "" {
+				swarm.PTP.PeersLock.Lock()
 				for _, peer := range swarm.PTP.NetworkPeers {
 					if peer.PeerLocalIP.String() == args.IP {
 						if peer.State == ptp.P_CONNECTED {
 							resp.ExitCode = 0
 							resp.Output = "Integrated with " + args.IP
+							swarm.PTP.PeersLock.Unlock()
+							runtime.Gosched()
 							return nil
 						}
 					}
 				}
+				swarm.PTP.PeersLock.Unlock()
+				runtime.Gosched()
 				resp.ExitCode = 1
 				resp.Output = "Not yet integrated with " + args.IP
 				return nil
 			} else {
 				resp.Output = "< Peer ID >\t< IP >\t< Endpoint >\t< HW >\n"
+				swarm.PTP.PeersLock.Lock()
 				for _, peer := range swarm.PTP.NetworkPeers {
 					resp.Output = resp.Output + peer.ID + "\t"
 					resp.Output = resp.Output + peer.PeerLocalIP.String() + "\t"
 					resp.Output = resp.Output + peer.Endpoint.String() + "\t"
 					resp.Output = resp.Output + peer.PeerHW.String() + "\n"
 				}
+				swarm.PTP.PeersLock.Unlock()
+				runtime.Gosched()
 			}
 		} else {
 			resp.Output = "Specified environment was not found: " + args.Hash

--- a/lib/crypto.go
+++ b/lib/crypto.go
@@ -35,7 +35,7 @@ func (c Crypto) EnrichKeyValues(ckey CryptoKey, key, datetime string) CryptoKey 
 	// Default value is +1 hour
 	ckey.Until = ckey.Until.Add(60 * time.Minute)
 	if err != nil {
-		Log(ERROR, "Failed to parse TTL. Falling back to default value of 1 hour")
+		Log(WARNING, "Failed to parse TTL. Falling back to default value of 1 hour")
 	} else {
 		ckey.Until = time.Unix(i, 0)
 	}

--- a/lib/peer.go
+++ b/lib/peer.go
@@ -60,7 +60,7 @@ func (np *NetworkPeer) Run(ptpc *PTPCloud) {
 		}
 		err := callback(ptpc)
 		if err != nil {
-			Log(ERROR, "Peer %s: %v", np.ID, err)
+			Log(WARNING, "Peer %s: %v", np.ID, err)
 		}
 		time.Sleep(time.Millisecond * 500)
 	}
@@ -323,7 +323,7 @@ func (np *NetworkPeer) TestConnection(ptpc *PTPCloud, endpoint *net.UDPAddr) boo
 	msg := CreateTestP2PMessage(ptpc.Crypter, "TEST", 0)
 	conn, err := net.DialUDP("udp4", nil, endpoint)
 	if err != nil {
-		Log(ERROR, "%v", err)
+		Log(DEBUG, "%v", err)
 		return false
 	}
 	ser := msg.Serialize()
@@ -340,7 +340,7 @@ func (np *NetworkPeer) TestConnection(ptpc *PTPCloud, endpoint *net.UDPAddr) boo
 		var buf [4096]byte
 		s, _, err := conn.ReadFromUDP(buf[0:])
 		if err != nil {
-			Log(ERROR, "%v", err)
+			Log(DEBUG, "%v", err)
 			conn.Close()
 			return false
 		}

